src/backends/dummy.rs

```diff
@@ -107,38 +132,52 @@ impl Iterator for DummyBlockIterator {
 
 #[cfg(test)]
 mod tests {
-    use super::DummyTracer;
+    use super::DummyThreadTracer;
     use ::test_helpers;
+    use backends::{TracerBuilder, BackendKind};
 
     #[test]
     fn test_basic_usage() {
-        test_helpers::test_basic_usage(DummyTracer::new());
+        test_helpers::test_basic_usage(DummyThreadTracer::new());
     }
 
     #[test]
     fn test_repeated_tracing() {
-        test_helpers::test_repeated_tracing(DummyTracer::new());
+        test_helpers::test_repeated_tracing(DummyThreadTracer::new());
     }
 
     #[test]
     fn test_already_started() {
-        test_helpers::test_already_started(DummyTracer::new());
+        test_helpers::test_already_started(DummyThreadTracer::new());
     }
 
     #[test]
     fn test_not_started() {
-        test_helpers::test_not_started(DummyTracer::new());
+        test_helpers::test_not_started(DummyThreadTracer::new());
     }
 
     #[test]
     fn test_block_iterator() {
-        use ::Tracer;
-        let mut tracer = DummyTracer::new();
+        use ::ThreadTracer;
+        let mut tracer = DummyThreadTracer::new();
         tracer.start_tracing().unwrap();
         let trace = tracer.stop_tracing().unwrap();
 
         // We expect exactly 0 blocks.
         let expects =  Vec::new();
         test_helpers::test_expected_blocks(trace, expects.iter());
     }
+
+    #[test]
+    fn test_unapplicable_config() {
+        let res = TracerBuilder::new()
+                                .backend(BackendKind::Dummy)
+                                .perf_aux_bufsize(1000) // Invalid.
+                                .build();
+        let expect = "perf_aux_bufsize config attribute does not apply to DummyThreadTracer";
+        match res {
```

Comment:
This could just be `assert_eq!(res.unwrap_err(), expect);`.

Reply:
Nice

Reply:
Huh, this is odd:
```
o method named `unwrap_err` found for type `std::result::Result<...>
```

And the functionality isn't in a trait...

Reply:
[Note that this would enforce this on our pubic API]

Reply:
No, we don't want to do that. Maybe it should be `assert_eq!(res.unwrap_err().to_string(), expect);`

Reply:
Well that would kill the need for `PartialEq` but not the need for `Debug`. You need to avoid using `unwrap_err` if you don't want `Debug`.

Reply:
```
impl<T, E> Result<T, E> where T: Debug {  // <--- Debug!
    pub fn unwrap_err(self) -> E
```

Reply:
I have no problem with `Debug`, but it's up to you as to whether you think the cost is worth it.

Reply:
I don't think there's any (performance) cost. It's just we force people implementing backends to implement `Debug`, which isn't strictly required.